### PR TITLE
expose stepNumber in locals

### PR DESCRIPTION
**Pull request by joefitter**
- added a stepData object to sessionModel - this contains the previous step and the steps journey
- a step is added to the steps journey after the index of the previous step
- invalidated steps are removed from the journey to keep an accurate index
- stepNumber is exposed to locals which is the index of the current step in the journey, this is 1-indexed

### CODE REVIEW

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @daniel-ac-martin would you mind having a 👀 when you get a sec please?

**Comment by JoeChapman** [comment]
Can we get some ES6 love in tha' house?!

File: lib/base-controller.js [hunk 1]
```diff
@@ -74,6 +74,34 @@ BaseController.prototype.getErrorStep = function getErrorStep(err, req) {
...
+  stepsJourney = _.intersection(stepsJourney, steps);
+  // add currentStep to journey if not
+  // already added
+  if (stepsJourney.indexOf(route) === -1) {
+    if (prevStep === undefined) {
+      stepsJourney.push(route);
+    } else {
+      // add current step after the prev step
+      // to preserve journey order
+      var prevIndex = stepsJourney.indexOf(prevStep);
```
Comment (JoeChapman) [inline lib/base-controller.js:21]: should be `let`

**Comment by joefitter** [comment]
@JoeChapman The rest of hof-controllers is still es5, I submitted a PR a few weeks back to ES6ify everything, probs needs rebasing now though, would suggest either merging this first, then rebasing, or rebasing that and merging, then rebasing this

File: lib/base-controller.js (hunk 1 above)
Comment (joefitter) [inline lib/base-controller.js:21]: still es5

**Comment by joefitter** [comment]
https://github.com/UKHomeOffice/hof-controllers/pull/28

**Comment by JoeChapman** [comment]
I see your point, but I really think any new work on code in our remit should be written using es6. What do you think?

**Comment by joefitter** [comment]
I think a mixture of both is bad, I think it should be all or nothing, at least per controller/file

**Comment by joefitter** [comment]
I'll rebase the ES6 branch then shall we look at merging that first?

**Comment by JoeChapman** [comment]
I do think it's ok to a bit at a time, at least an entire file at a time. There is no reason afaik why we should need to rewrite everything in es6

**Comment by joefitter** [comment]
Sure, my point is it has already been done but not merged. Fine to rewrite the base controller in ES6 despite this though if we are not wanting to merge the ES6 refactor

**Comment by JoeChapman** [comment]
Oh, right, ok, I understand you. no, no, let's rebase the ES6 pr

**Comment by JoeChapman** [comment]
ok LGTM.

**Comment by joefitter** [comment]
if https://github.com/UKHomeOffice/hof-controllers/pull/28 is merged first I will rebase against that

**Comment by joefitter** [comment]
@easternbloc @JoeChapman

**Comment by joefitter** [comment]
@JoeChapman @easternbloc how do we feel about this? Step labels is still a grey area I think due to totalSteps changing when we fork. We discussed just showing the current step in relation to completed steps which is achieved by this PR

**Comment by JoeChapman** [comment]
LGTM, @easternbloc?

**Comment by joefitter** [comment]
@easternbloc @daniel-ac-martin can I get a 👍  plz?

**Pull request closed without merging**